test/test_nn.py

```diff
@@ -1919,6 +1919,41 @@ def test_Conv2d_groups_nobias(self):
             self.assertEqual(m.weight.grad.data,
                              torch.cat([m1.weight.grad.data, m2.weight.grad.data], 0))
 
+    # Very similar to test_Conv2d_naive_groups but with special care to handle
+    # the number of groups == number of input channels
+    @unittest.skipIf(not TEST_CUDA, 'CUDA not available')
+    def test_Conv2d_depthwise_naive_groups(self):
```

Comment:
Does this test datatypes other than float? If not, can you please add half and double tests?

Reply:
Will do.